app/src/main/java/com/android/shelfLife/ui/newoverview/NewFoodDisplayComponents.kt

```diff
@@ -0,0 +1,179 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color.Companion.LightGray
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil3.compose.AsyncImage
+import com.android.shelfLife.model.foodFacts.FoodUnit
+import com.android.shelfLife.model.newFoodItem.FoodItem
+import com.android.shelfLife.model.newFoodItem.FoodStatus
+import com.android.shelfLife.model.newOverview.OverviewScreenViewModel
+import com.android.shelfLife.ui.utils.getExpiryMessageBasedOnDays
+import com.android.shelfLife.ui.utils.getProgressBarState
+import com.google.firebase.Timestamp
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+/**
+ * Composable function to display the list of food items
+ *
+ * @param foodItems The list of food items to display
+ */
+@Composable
+fun ListFoodItems(
+    foodItems: List<FoodItem>,
+    overviewScreenViewModel: OverviewScreenViewModel,
+    onFoodItemClick: (FoodItem) -> Unit,
+    onFoodItemLongHold: (FoodItem) -> Unit
+) {
+  if (foodItems.isEmpty()) {
+    Box(
+        modifier = Modifier.fillMaxSize().testTag("NoFoodItems"),
+        contentAlignment = Alignment.Center) {
+          Text(text = "No food available")
+        }
+  } else {
+    // Display the full list
+    LazyColumn(modifier = Modifier.fillMaxSize().testTag("foodItemList")) {
+      items(foodItems) { item ->
+        // Call a composable that renders each individual to-do item
+        FoodItemCard(
+            foodItem = item,
+            overviewScreenViewModel = overviewScreenViewModel,
+            onClick = { onFoodItemClick(item) },
+            onLongPress = { onFoodItemLongHold(item) })
+      }
+    }
+  }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun FoodItemCard(
+    foodItem: FoodItem,
+    overviewScreenViewModel: OverviewScreenViewModel,
+    onClick: () -> Unit = {},
+    onLongPress: () -> Unit = {}
+) {
+  val selectedItems by overviewScreenViewModel.multipleSelectedFoodItems.collectAsState()
+  val isSelected = selectedItems.contains(foodItem)
+  val cardColor =
+      if (isSelected) MaterialTheme.colorScheme.primaryContainer
+      else MaterialTheme.colorScheme.background
+  val elevation = if (isSelected) 16.dp else 8.dp
+  val expiryDate = foodItem.expiryDate
+  val currentDate = Timestamp.now()
+
+  // Calculate time remaining in days
+  val timeRemainingInDays =
+      expiryDate?.let { ((it.seconds - currentDate.seconds) / (60 * 60 * 24)).toInt() } ?: -1
+
+  // Get progress bar state
+  val (progress, progressBarColor) = getProgressBarState(timeRemainingInDays)
+
+  // Get formatted expiry date and message
+  val formattedExpiryDate =
```

Comment:
i think there are already util functions that do this kind of formating in an util class